client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -472,6 +469,29 @@ export default {
       this.updateBlocks(block)
     },
 
+    untabBlock (block) {
+      const { isTabbed, where } = this.checkTabbedState(block)
+      if (isTabbed) {
+        const { tabs } = where.options
+        const index = tabs.findIndex(({ blockID }) => blockID === fetchID(block))
+        tabs.splice(index, 1)
+      }
+    },
+
+    checkTabbedState (block) {
+      const tabs = this.blocks.filter(({ kind }) => kind === 'Tabs')
```

Comment:
Instead of going through it twice, how about you just do a blocks.forEach, and check if its a tab block, if it is then check if your block is in it, if it is add it to the array of indexes where you keep the location of blocks that have your block. 
And in untab just walk through the indexes and splice your block.